component/src/main/java/io/siddhi/extension/execution/file/FileArchiveExtension.java

```diff
@@ -0,0 +1,422 @@
+                ),
+                @Example(
+                        syntax = "InputStream#file:archive('/User/wso2/to_be_archived', " +
+                                "'/User/wso2/archive_destination/file', '', '', 'false')",
+                        description = "Archives to_be_archived folder excluding the sub-folders in zip format and " +
+                                "stores in archive_destination folder as file.tar."
+                )
+        }
+)
+public class FileArchiveExtension extends StreamProcessor<State> {
+    private static final Logger log = Logger.getLogger(FileArchiveExtension.class);
+    private Pattern pattern;
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater,
+                           State state) {
+        while (streamEventChunk.hasNext()) {
+            StreamEvent streamEvent = streamEventChunk.next();
+            String uri = (String) attributeExpressionExecutors[0].execute(streamEvent);
+            String destinationDirUri = (String) attributeExpressionExecutors[1].execute(streamEvent);
+            boolean excludeSubdirectories = false;
+            String regex = "";
+            String archiveType = ZIP_FILE_EXTENSION;
+            if (attributeExpressionExecutors.length >= 3) {
+                archiveType = (String) attributeExpressionExecutors[2].execute(streamEvent);
+            }
+            if (attributeExpressionExecutors.length >= 4) {
+                regex = (String) attributeExpressionExecutors[3].execute(streamEvent);
+            }
+            if (attributeExpressionExecutors.length == 5) {
+                excludeSubdirectories = (Boolean) attributeExpressionExecutors[4].execute(streamEvent);
+            }
+            pattern = Pattern.compile(regex);
+            if (archiveType.compareToIgnoreCase(ZIP_FILE_EXTENSION) == 0) {
+                File sourceFile = new File(uri);
+                List<String> fileList = new ArrayList<>();
+                generateFileList(uri, sourceFile, fileList, excludeSubdirectories);
+                zip(uri, destinationDirUri, fileList);
+            } else {
+                try {
+                    if (archiveType.compareToIgnoreCase(TAR_FILE_EXTENSION) == 0) {
+                        addToTarArchiveCompression(
+                                getTarArchiveOutputStream(destinationDirUri), new File(uri), uri);
+                    } else {
+                        throw new SiddhiAppRuntimeException("Unsupported archive type: " + archiveType);
+                    }
+                } catch (IOException e) {
+                    throw new SiddhiAppRuntimeException("Exception occurred when archiving " + uri, e);
+                }
+            }
+            Object[] data = {true};
```

Comment:
Why the status of the file archive operation is always `true`? If it fails, are we just breaking the flow with `SiddhiAppRuntimeException`?

Reply:
@grainier we have removed true as well since if the event returns from the function t means the operation is successful. Added SiddhiAppRuntimeException to break the flow since we can handle it with error streams afterwords.